### PR TITLE
Update Testing For Team 4

### LINKED ISSUE #616: [Team 4] Update Tests to Ensure Line Coverage for All Classes Created By Team 4
# **Description**

This task is to ensure that all the classes written and used by team 4 has been properly covered. In addition, the documentation will be updated accordingly to match the new test files.
 
# **Milestones**
- [x] NPCFactory Tests
- [x] BaseFriendlyEntityConfig Tests
- [x] DialogueBoxService Tests
- [ ] Documentation of all test files
 
# **Member**
- Julian Wong (@Jul-Wong)

### PR DESCRIPTION
**Pull request by Jul-Wong**
# Description

Closes #616 ensuring proper coverage of classes written and used by team 4. Makes no change to functionality and only changes test classes.

### CODE REVIEW

File: .github/pull_request_template.md
Comment (LucasRyall) [inline .github/pull_request_template.md:1]: Do not remove this file.

File: source/core/src/main/com/csse3200/game/minigames/maze/components/npc/MazePlayerAnimationController.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
+package com.csse3200.game.minigames.maze.components.npc;
+
+import com.csse3200.game.components.Component;
+import com.csse3200.game.rendering.AnimationRenderWithAudioComponent;
+
+/**
+ * This class listens to events relevant to the player's state and plays the appropriate animation
+ * when an event is triggered.
+ */
+public class MazePlayerAnimationController extends Component {
```
Comment (jaden-z-fng) [inline source/core/src/main/com/csse3200/game/minigames/maze/components/npc/MazePlayerAnimationController.java:10]: Is this class ever used? IntelliJ is saying that it isn't.

Comment (jaden-z-fng) [inline source/core/src/main/com/csse3200/game/minigames/maze/components/npc/MazePlayerAnimationController.java:10]: Is this class ever used? IntelliJ is saying that it isn't.

**Review by jaden-z-fng — COMMENTED** [review]
Just that little thing about MazePlayerAnimationController, otherwise looks good

**Review by jaden-z-fng — APPROVED** [review]

**Pull request merged**